llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -2221,7 +2226,7 @@ static bool eliminateDeadStores(Function &F, AliasAnalysis &AA, MemorySSA &MSSA,
           continue;
         LLVM_DEBUG(dbgs() << "DSE: Remove Dead Store:\n  DEAD: " << *DeadI
                           << "\n  KILLER: " << *KillingI << '\n');
-        State.deleteDeadInstruction(DeadI);
+        State.deleteDeadInstruction(DeadI, &Deleted);
```

Comment:
Would it be possible to check if all uses in the loop pass Deleted, e.g. add an assert that checks if we add as many elements to `SkipStore` as we do to `Deleted` after the loop?

Reply:
I've added an assertion -- does that match what you had in mind?

Reply:
Thanks!